File: app/src/main/java/com/jx/androiddemo/presenter/BaseRxPresenter.java
package com.jx.androiddemo.presenter;

import com.jx.androiddemo.BaseView;

import java.lang.ref.Reference;
import java.lang.ref.WeakReference;

/**
 * Created by Administrator on 2017/3/28.
 */

public abstract class BaseRxPresenter<T extends BaseView> implements BasePresenter<T>
{
    protected Reference<T> mView;

    public abstract void doDispose();

    @Override
    public void attachView(T view)
    {
        mView = new WeakReference<>(view);
    }

    protected T getView()
    {
        if (mView == null)
        {
            return null;
        }
        return mView.get();//获取View
    }

    public boolean isViewAttached()//判断是否与View建立了关联
    {
        return mView != null && mView.get() != null;
    }

    @Override
    public void detachView()
    {
        if (mView != null)
        {
            mView.clear();
            mView = null;
        }
    }
}

File: app/src/main/java/com/jx/androiddemo/presenter/BaseRxPresenter.kt
package com.jx.androiddemo.presenter

import java.lang.ref.Reference
import java.lang.ref.WeakReference

abstract class BaseRxPresenter<T> : BasePresenter<T> {
    var mView: Reference<T>? = null
    abstract fun doDispose()
    open fun getView(): T? = mView?.get()
    open fun isViewAttached(): Boolean = null != mView?.get()

    override fun attachView(view: T) {
        mView = WeakReference(view)
    }

    override fun detachView() {
        mView?.clear()
        mView = null
    }
}